File: 137-single-number-ii/single-number-ii.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = 0
        for i in range(32):
            bit_sum = sum((num >> i) & 1 for num in nums)
            result |= (bit_sum % 3) << i

        if result & (1 << 31):
            result -= 1 << 32

        return result

File: 137-single-number-ii/single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        int[] bitCount = new int[32];

        for (int num : nums) {
            for (int i = 0; i < 32; i++) {
                bitCount[i] += (num >> i) & 1;
                bitCount[i] %= 3;
            }
        }

        int result = 0;

        for (int i = 0; i < 32; i++) {
            result |= (bitCount[i] << i);
        }

        return result;
    }
}